hardware_interface/source/hyperion.c

```diff
@@ -237,40 +247,149 @@ void hyperion_config_2_value(config_2_panel_t panel, config_2_channel_type_t cha
     case CONFIG_3_CHANNEL_TEMP_1:
         adc_init(slave_addr, 1 << 7);
         adc_get_raw(slave_addr, &data, &ch);
+        adc_get_raw(slave_addr, &data, &ch);
         *param = adc_calculate_sensor_temp(&data, ADC_VREF);
         break;
 
     case CONFIG_3_CHANNEL_TEMP_2:
         adc_init(slave_addr, 1 << 6);
         adc_get_raw(slave_addr, &data, &ch);
+        adc_get_raw(slave_addr, &data, &ch);
         *param = adc_calculate_sensor_temp(&data, ADC_VREF);
         break;
 
     case CONFIG_3_CHANNEL_PD_1:
         adc_init(slave_addr, 1 << 5);
         adc_get_raw(slave_addr, &data, &ch);
+        adc_get_raw(slave_addr, &data, &ch);
         *param = adc_calculate_sensor_pd(&data, ADC_VREF);
         break;
 
     case CONFIG_3_CHANNEL_PD_2:
         adc_init(slave_addr, 1 << 4);
         adc_get_raw(slave_addr, &data, &ch);
+        adc_get_raw(slave_addr, &data, &ch);
         *param = adc_calculate_sensor_pd(&data, ADC_VREF);
         break;
 
     case CONFIG_3_CHANNEL_VOLT:
         adc_init(slave_addr, 1 << 3);
         adc_get_raw(slave_addr, &data, &ch);
+        adc_get_raw(slave_addr, &data, &ch);
         *param = adc_calculate_sensor_voltage(&data, ADC_VREF);
         break;
 
     case CONFIG_3_CHANNEL_CURR:
         adc_init(slave_addr, 1 << 2);
         adc_get_raw(slave_addr, &data, &ch);
+        adc_get_raw(slave_addr, &data, &ch);
         *param = adc_calculate_sensor_current(&data, ADC_VREF);
         break;
-    
+
     default:
         break;
     }
- }
+}
+
+void Hyperion_getHK(Hyperion_HouseKeeping* hyperion_hk) {
```

Comment:
Has this code been tested on hardware?

Reply:
No I have not tested this function but given that the other hyperion functions are working (I tested them before) so I assume there would be no problem, because we are just calling them.

Reply:
Note that this function would only work on Hyperion (3U) not on Hyperion 2U. We will probably need a similar function to correctly call the housekeeping on the other satellites.